### DISCUSSION TITLE
Use the value of a variable in a string

### DISCUSSION THREAD

**Discussion by rondnelson99** [vhelin/wla-dx#457] [Q&A] [answered]
I'm back, and once again I'm trying to figure out what WLA's best equivalent is to a feature of RGBDS. My project sprinkles level data throughout the ROM in a very unconventional way, and I have this loop that is supposed to generate it:
```
.rept NUM_ROWS index ROW
    .org ROW << 8
    .section "level_row_{ROW}" FORCE
        .rept NUM_COLS index COL
            .arraydb level_vert COL+ROW*NUM_COLS
            .arraydb level_horiz COL+ROW*NUM_COLS
        .endr
    .ends
.endr
```
The important part here is that the data is spread throughout several sections that are generated in the loop. I understand that the sections need to have unique names, so my goal was to include the row number in the name. With RGBDS, I could use [symbol interpolation](https://rgbds.gbdev.io/docs/master/rgbasm.5#Symbol_interpolation) to accomplish this. How would I do this in WLA? I understand that `\@` is the intended way to make things unique in REPEAT blocks, but "level_row_\@" isn't valid either.
Thanks, 
Ron

**Comment by vhelin** [comment, marked as answer]
\\@ works only in .MACRO, but you could do something like this:

```
        .define NUM_ROWS 8
        .define NUM_COLS 8

        .macro CreateSection ARGS ROWm
          .section "level_row_\1" FORCE
          .rept NUM_COLS index COL
            .dw COL+ROWm*NUM_COLS
            .dw COL+ROWm*NUM_COLS
          .endr
        .endm

        .rept NUM_ROWS index ROW
          .org ROW << 8
          CreateSection ROW
          .ends
        .endr
```

**Reply by rondnelson99** [reply]
Interesting, thanks. So basically you're saying that \\@, \1, etc are expanded when used in strings, but all other variables are not. Is that correct? 

BTW, what is the term that WLA uses for its build-time variables? I have seen other assemblers that call them "symbols" or "labels" (even when they do not reference a location in the ROM). What does WLA call them?

**Reply by vhelin** [reply]
There is no expansion for defines, only .MACRO arguments. You can always use a define as a .MACRO argument... This is where WLA-DX could be better, perhaps in the future I'll add support for global variable expansion in strings, if no-one else does that... Making a function for that, like defined() or asc(), might be the way to go?

Build-time variables? They are either .MACRO arguments or definitions, I guess?

**Reply by rondnelson99** [reply]
Ok, thanks! I suppose defines or definitions is the word I was looking for. I agree that global variable expansion is a useful feature that WLA would benefit from. However, the macro workaround works perfectly fine.

**Comment by vhelin** [comment]
If you now grab the latest sources, you can do a substitution as follows:

label_{COUNTER}:
    .db { "COUNTER={COUNTER}" }
